src/library/Uno.Material/Controls/Ripple.cs

```diff
@@ -138,16 +136,18 @@ protected override void OnPointerPressed(PointerRoutedEventArgs e)
 
 		private void StartRippling(object _, PointerRoutedEventArgs e)
 		{
-			// workaround for uno#5033 nested `Ripple` controls all ripples, intead of just the innermost (side effect of workaround material#373)
+			// workaround for uno#5033 and uno#5874 nested `Ripple` controls all ripples, instead of just the innermost (side effect of workaround material#373)
 #if NETSTANDARD2_0
-			// Because the event couldve been already handled by other control, e.Handled cannot be used here.
-			// RippleHandled is used instead to indicate an ripple has occured, to prevent the next on the line to ripple again.
-			if (e.OriginalSource is DependencyObject os)
-			{
-				if ((bool)os.GetValue(RippleHandledProperty)) return;
+			// Because the event could've been already handled by other control, e.Handled cannot be used here.
+			// FrameId is used instead to verify if a ripple has already occurred with this Id, to prevent the next on the line to ripple again.
+			var currentFrameId = e.GetCurrentPoint(null).FrameId;
 
-				os.SetValue(RippleHandledProperty, true);
+			if (currentFrameId <= _lastHandledFrameId)
+			{
+				return;
 			}
+
+			_lastHandledFrameId = currentFrameId;
 #endif
 
 			var point = e.GetCurrentPoint(this);
```

Comment:
Just noticed that we do a `e.GetCurrentPoint` also here. We should merge is with `var currentFrameId = e.GetCurrentPoint(null).FrameId;` on line 143 ... perf is everywhere :)

Reply:
Thanks a lot @dr1rrb for noticing :) ! I updated the code.